GoogleForCreators/web-stories-wp
# Prepublish Checklist:  Thumbnail Component

## Context

Adds thumbnail component for use in new card system that prepublish checklist will use. 

## Summary

The prepublish checklist is getting updated to use a card system. At the core of this card system are thumbnails that render each element type or page depending on the issue the checklist is displaying. This PR just gets that UI ready, it lives entirely in storybook. 

## Relevant Technical Choices

Rather than reinventing the wheel I decided to pull from the Layer Panel and the Carousel for how the elements and pagePreviews are getting created and rendered. This also makes the UI consistent.  We have access to the same element/page details in the checklist, so when thumbnails are created they'll be done either like this: 
```
const { LayerIcon } = getDefinitionForType(layer.type);
const LayerThumbnail = <LayerIcon element={layer} />;
...
<Thumbnail  displayBackground={LayerThumbnail} />);
```
Or like this, using the `page` of a story and then just override what styled components is going to render it as and give it the dimensions of the thumbnail (which are the same as the carousel but still need to be passed in).
```
const PageThumbnail = (
  <PagePreview
    page={page}
    width={THUMBNAIL_DIMENSIONS.WIDTH}
    height={THUMBNAIL_DIMENSIONS.HEIGHT}
    as="div"
  />
);

...
<Thumbnail  displayBackground={PageThumbnail} ... />;
```

That's the only big thing to note here. Otherwise, I did decide to make the scrim it's own layer on the thumbnail buttons because it got weird dealing with the different layers - icons need to have a full opacity (like if there's an error) and sometimes (for Page type thumbnails) since I'm using the PagePreview the shapes could have a higher index which needs to be over powered. And then also there could be tooltips for hover (they don't actually have any further interaction available to them because they're part of the same button, confirmed this behavior with design). Anyway, with the opacity of the scrim and the borders it was just more straight forward to make it a layer on top of the button.

And since the tooltip is only accessible on hover, we'll just need to make sure that the aria label on the button is getting the necessary info. Or we can make a follow up ticket for tooltips to adjust showing the tooltip nested within a button but that feels sketchy. Only time this is a thing right now is video optimization, doesn't feel worth the time to dig deeper into ATM.  

## To-do

- [ ] Confirm how to render horizontal images in thumbnails. Should we cover or contain? If we contain then there's no default border until interaction and it's kinda weird. 
- [ ] Confirm with design the idea of keeping thumbnails consistent with layers panel instead of the "lorem ipsum" text, since some text issues are that there is no text and the 'T' Text icon is already in play to mean text elsewhere in the editor (plus then we can reuse code!) 

## User-facing changes

(Storybook only) 
<img width="431" alt="Screen Shot 2021-06-14 at 3 06 18 PM" src="https://user-images.githubusercontent.com/10720454/121966818-251b6080-cd24-11eb-80ea-fcb4b85c3cc1.png">

![thumbnails in storybook](https://user-images.githubusercontent.com/10720454/121966842-2b114180-cd24-11eb-9606-557c913f4b9a.gif)


## Testing Instructions

Spin up storybook and go to stories-editor/components/thumbnail  and interact with thumbnails. Compare to figma: https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=8378%3A3528 . Note the intentional separation from the Text thumbnail to reuse the layers panel icon for text (see above for note and todo item). Should have focus, hover and an error state available. 

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #7913
